src/App.tsx

```diff
@@ -16,47 +16,82 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const visibleGoods = [...goodsFromServer];
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortBy, setSortBy] = useState(0);
+
+  if (sortBy === 1) {
+    visibleGoods.sort();
+  }
+
+  if (sortBy === 2) {
+    visibleGoods.sort((g1, g2) => {
+      return g1.length - g2.length;
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortBy(1)}
           type="button"
-          className="button is-info is-light"
+          className={
+            sortBy === 1
+              ? 'button is-info'
+              : 'button is-info is-light'
+          }
```

Comment:
use [classnames](https://www.npmjs.com/package/classnames) library to add classes conditionally